tools/clang/test/HLSLFileCheck/pix/DebugRayGen.hlsl

```diff
@@ -0,0 +1,75 @@
+// RUN: %dxc -T lib_6_3 %s | %opt -S -hlsl-dxil-debug-instrumentation,parameter0=10,parameter1=20,parameter2=30 | %FileCheck %s
+
+
+// Just check that the selection prolog was added:
+
+// CHECK:  %RayX = call i32 @dx.op.dispatchRaysIndex.i32(i32 145, i8 0)
+// CHECK:  %RayY = call i32 @dx.op.dispatchRaysIndex.i32(i32 145, i8 1)
+// CHECK:  %RayZ = call i32 @dx.op.dispatchRaysIndex.i32(i32 145, i8 2)
+
+// CHECK: %CompareToThreadIdX
+// CHECK: %CompareToThreadIdY
+// CHECK: %CompareToThreadIdZ
+// CHECK: %CompareXAndY
+// CHECK: %CompareAll
```

Comment:
I'm a bit surprised that you can expect these register names to persist. Maybe I'm missing something?

Reply:
You are of course correct. This tickled my mind when I coded it and I ignored the tickle. Bad Jeffrey.